Sources/Core/GTMSessionFetcher.m

```diff
@@ -2066,6 +2068,7 @@ - (void)stopFetchReleasingCallbacks:(BOOL)shouldReleaseCallbacks {
         }
       }
     }
+    shouldStopFetcher = !_stopFetchingTriggersCompletionHandler || !_userStoppedFetching;
```

Comment:
Why is this one `!_userStoppedFetcher`?

If `stopFetchingTriggersCompletionHandler` is `NO`, and `stopFetching` is called, then `[service fetcherDidStop:self]` on line 2090 will be skipped, so what actually does that work now?

At first reread, should this be `!_stopFetchingTriggersCompletionHandler && _userStoppedFetching`?  That would make the code continue the current behavior before adding `_stopFetchingTriggersCompletionHandler`.

Reply:
This change hangs the `testFetcherService` and other tests. I'll investigate.

Reply:
After further investigation, I believe the logic was correct. I inverted the condition and it's hopefully easier to understand.

Reply:
Thanks, yea, this does read easier.